packages/calendar/src/Calendar.js

```diff
@@ -7,111 +7,146 @@
  * file that was distributed with this source code.
  */
 import React from 'react'
-import { Container, SvgWrapper } from '@nivo/core'
+import { SvgWrapper, useTheme, useDimensions, withContainer, useValueFormatter } from '@nivo/core'
 import { BoxLegendSvg } from '@nivo/legends'
-import { setDisplayName } from 'recompose'
-import { CalendarPropTypes } from './props'
-import enhance from './enhance'
+import { CalendarPropTypes, CalendarDefaultProps } from './props'
+
 import CalendarYearLegends from './CalendarYearLegends'
 import CalendarMonthPath from './CalendarMonthPath'
 import CalendarMonthLegends from './CalendarMonthLegends'
+
+import { useMonthLegends, useYearLegends, useCalendarLayout, useDays, useColorScale } from './hooks'
 import CalendarDay from './CalendarDay'
 
 const Calendar = ({
-    colorScale,
-
-    margin,
+    margin: partialMargin,
     width,
     height,
-    outerWidth,
-    outerHeight,
 
-    yearLegends,
+    align,
+    colors,
+    colorScale,
+    data,
+    direction,
+    emptyColor,
+    from,
+    to,
+    minValue,
+    maxValue,
+    valueFormat,
+    legendFormat,
+
     yearLegend,
+    yearLegendOffset,
+    yearLegendPosition,
+    yearSpacing,
 
-    monthLegends,
-    monthLegend,
-    monthBorderWidth,
     monthBorderColor,
+    monthBorderWidth,
+    monthLegend,
+    monthLegendOffset,
+    monthLegendPosition,
+    monthSpacing,
 
-    daySpacing,
-    dayBorderWidth,
     dayBorderColor,
-
-    theme,
+    dayBorderWidth,
+    daySpacing,
 
     isInteractive,
-    tooltipFormat,
     tooltip,
     onClick,
+    onMouseEnter,
+    onMouseLeave,
+    onMouseMove,
 
     legends,
-
-    months,
-    days,
 }) => {
+    const theme = useTheme()
+    const { margin, innerWidth, innerHeight, outerWidth, outerHeight } = useDimensions(
+        width,
+        height,
+        partialMargin
+    )
+    let { months, years, days } = useCalendarLayout({
```

Comment:
```suggestion
    const { months, years, ...rest } = useCalendarLayout({
```

Avoid reassignment when needed, and we can make this a `const`.